src/test/fuzz/bech32.cpp

```diff
@@ -29,8 +29,9 @@ FUZZ_TARGET(bech32)
     std::vector<unsigned char> input;
     ConvertBits<8, 5, true>([&](unsigned char c) { input.push_back(c); }, buffer.begin(), buffer.end());
 
-    if (input.size() + 3 + 6 <= 90) {
-        // If it's possible to encode input in Bech32(m) without exceeding the 90-character limit:
+    // Input data part + 3 characters for the HRP and separator (bc1) + the checksum characters
+    if (input.size() + 3 + bech32::CHECKSUM_SIZE <= bech32::CharLimit::BECH32) {
```

Comment:
Per https://github.com/bitcoin/bitcoin/pull/30596#discussion_r1705432774, it would be nice to not hardcode the HRP at all (or it's size) but I think it should be a separate and larger change. For the separator, I don't think it makes much difference if we use a variable or comment to mention what the value is for.

Reply:
I'm fine either way